product-service/src/functions/createProduct/handler.ts

```diff
@@ -0,0 +1,49 @@
+import { APIGatewayProxyEvent } from 'aws-lambda';
+import { formatJSONResponse } from '@libs/api-gateway';
+import { middyfy } from '@libs/lambda';
+import { headers } from '@libs/headers';
+import * as productService from '@services/product.service';
+import { createProductSchema } from '@libs/schemaValidations';
+
+const createProduct = async (event: APIGatewayProxyEvent) => {
+  try {
+    console.log('createProduct Lambda: Event: ', event);
+
+    const data = JSON.parse(JSON.stringify(event.body));
+    console.log('createProduct Lambda: data: ', data);
+
+    try {
+        await createProductSchema.validate(data);
+    } catch(err) {
+        return formatJSONResponse({
+            statusCode: 400,
+            headers,
+            result: {
+                ErrorName: err.name,
+                Errors: err.errors
+            },
+        });
+    }
+
+    const newProduct = await productService.createProduct(data);
+    console.log('createProduct Lambda: new product: ', newProduct);
+
+    const res = formatJSONResponse({
+      statusCode: 200,
+      headers,
+      message: 'new product was sucessfully created',
+      result: newProduct,
+  });
+    console.log('createProduct Lambda: response: ', res);
+    return res;
+  } catch (err) {
+    console.error('createProduct Lambda: Error Encountered: ', err);
+    return formatJSONResponse({
```

Comment:
As I can see this code won't be reached in case an error is occurred on `db.put()` step, as the last one it's wrapped in `try..catch` block and does't throw anything